express-notes-api/index.js

```diff
@@ -0,0 +1,135 @@
+const fs = require('fs');
+const express = require('express');
+const app = express();
+const notes = require('./data.json');
+
+app.get('/api/notes', function (req, res) {
+  const outputArray = [];
+  const notesData = notes.notes;
+  for (const property in notesData) {
+    outputArray.push(notesData[property]);
+  }
+  res.json(outputArray);
+});
+
+app.use(express.json());
+
+app.get('/api/notes/:id', function (req, res) {
+  const currentId = req.params.id;
+  const notesData = notes.notes;
+  let foundId = false;
+  const integerId = parseInt(currentId);
+
+  if (integerId < 0 || isNaN(integerId)) {
+    res.status(400).send('"error": "id must be a positive integer"');
+  } else {
+    for (const property in notesData) {
+      if (property === currentId) {
+        foundId = true;
+        res.json(notesData[property]);
+        fs.writeFile('data.json');
+        break;
+      }
+    }
```

Comment:
You don't need a loop to find a value in an object if you know the property name. You can just do:

```JAVASCRIPT
if(!notesData[currentId]) {
  return res.status(404).json({error: `cannot find note with id ${currentId}`});
}
```